test/case/ietf_system/ssh_key_authentication/test.py

```diff
@@ -0,0 +1,100 @@
+aUSLET3w+zKfK8HYiHxry6Jf+dcts9PQyxpCiEW3RoU65nwnq1pG7XAdmT/DeGKSQqXkdn
+mXvgZkrQYZ+4HW2veQlHzwwEIAhMTscS+QiOX/AAAAAwEAAQAAAQAOYsviyc9ZaF7ODWiJ
+Mg5zjcdFAaVHLKQlEagJdOQq9GNTguC5cTHXJQoK35nIQKGDIjSpUGn9jN+FVuU4XVsN8d
+JAbSgjqYdExzZNrVzLrbdvP7MsQrFNTwpcOaK37mhqcEQW27jzHNUB1f6pVwIOqGlmWcd6
+/unO/uhI37omyfCkZT7lU4E+iCGrbIiwmutT2KmUJ6FZOedDl7XbwyUZMqmtjOGbN+0Hc3
+ahtA79+C4MaIptGVr10c85WLz/Rh9oN9tDYZRqhU5cARUd9LyEeKF9R7L+kCq4b2zuEqNI
+BSYLGIu859f+IEXeA2Hq965tCTMMgwauMAS1If9R+rlRAAAAgHeoqdL7d3FX1iijZWSBua
+Bran5mNdcGzf/8SD25WRMUPhsoqHOvtDWuOkop6nAu/9GGigJaBICuDxJz3KN1Hhpuer+g
+/x7+Dgwd5mW41hXOTezp2hGO98OwHLJpmohCz1NIZIHz0h6NZbRZ4vozMtYBs1aq1Qv6h2
+DLsxLMLY7FAAAAgQDKkDQXxiXr7Bv7UJaYA4BZAg6qPMvW943mOreLl0uO8urE3mt2MnVy
+Et1r2dEhxwqsWrDj5rsBVSXrg/8kXYHrqAgldZ3TxEOCU8xVV4RAzxfE0uFwdawZHdsIJL
+JWwNnQ8TdWXPZze0lE9R5b2E4EvJilHvD13pdcIN+hyvGE0QAAAIEAyY+JAtJjDS6wW07E
+eTVndLHpDiTZO+tCIP+OO45uYA/8O+IMc+wvIsHCpZC7e3bskg6z2gt5B0DwSnf2Q4zXxP
+OFXdOBTFO8XcgWKRo9BIPV6BNH9qrx0Z0m5G45rY6SE9c5Ypv0ExjXRZ/iPWLBSarsv1fF
+lrH5aNT6hzZKsc8AAAAMcm9vdEBpbmZhbXkwAQIDBAUGBw==
+-----END OPENSSH PRIVATE KEY-----
+"""       
+        
+        public_key_alg = "ssh-rsa"
+        public_key_data = "AAAAB3NzaC1yc2EAAAADAQABAAABAQCffMvdK1VBkuw/12etfEr89Piba2NbQz7YY3ol2IhRnIoj0oodZp/a4c11c1Q+jQkbCDKPWB45w3ZMR2Q93XXeY72qtLzvCBLOzeCkX3czMAJLLxznl3EXmDnTAWetE1nUmByLLOvKdw1CPFiS9tushwdfj6iOfex3HvQwltV5i/rQz0h7JeSwEQ0Xn6sBgDxWlD6TtwYl4ch0zHoxRHCjqdQig7NMonhaF53BMKRYaUSLET3w+zKfK8HYiHxry6Jf+dcts9PQyxpCiEW3RoU65nwnq1pG7XAdmT/DeGKSQqXkdnmXvgZkrQYZ+4HW2veQlHzwwEIAhMTscS+QiOX/"
+
+    with test.step("Create a guest user on the target device"):
+        USER = "guest"
+
+        target.put_config_dict("ietf-system", {
+            "system": {
+                "authentication": {
+                    "user": [
+                        {
+                            "name": USER,
+                            "authorized-key": [
+                                {
+                                    "name":"guest-ssh-key",
+                                    "algorithm":public_key_alg,
+                                    "key-data":public_key_data
+                                }
+                            ],
+                            "infix-system:shell":"bash"
+                        }
+                    ]
+                }
+            }
+        })
+    
+    with test.step("Write private key to a temporary file"):
+        with tempfile.NamedTemporaryFile(delete=False, mode='w') as key_file:
+            key_file.write(private_key)
+            key_file_name = key_file.name
+
+        os.chmod(key_file_name, 0o600)
+
+    with test.step("Verify it is possible to fetch syslog data using public key"):
+        address = target.get_mgmt_ip()
+        syslog_file = "./syslog_copy"
+        result = subprocess.run(f"scp -q -o StrictHostKeyChecking=no -i {key_file_name} {USER}@[{address}]:/var/log/syslog {syslog_file}", shell=True)
```

Comment:
This could be refactored into a framework helper function so other test cases can reuse it easier. Maybe even directly in the ssh class?

I give you two options, you decide:

 1. Refactor now, with the added benefit of making the temporary file cleanup part of that common API
 2. Add a TODO comment for future readers, i.e., postpone for later